ports/raspberrypi/common-hal/busio/UART.c

```diff
@@ -42,13 +42,16 @@ void never_reset_uart(uint8_t num) {
     uart_status[num] = STATUS_NEVER_RESET;
 }
 
+static void pin_check(const uint8_t uart, const mcu_pin_obj_t *pin, const uint8_t pin_type) {
+    if (!(((pin->number % 4) == pin_type) && ((((pin->number + 4) / 8) % NUM_UARTS) == uart))) {
```

Comment:
I tested on a Metro RP2040 and discovered that now no combination of pins was valid. This is because if any pins are `NULL` (e.g. `cts` or `rts`, in the typical case), this will raise "invalid pin". I think this will fix it.

```suggestion
    if (pin == NULL) {
        return;
    }
    if (!(((pin->number % 4) == pin_type) && ((((pin->number + 4) / 8) % NUM_UARTS) == uart))) {
```